mapper/orm/src/main/java/org/hibernate/search/mapper/orm/automaticindexing/filter/impl/HibernateOrmAutomaticIndexingTypeFilterContext.java

```diff
@@ -0,0 +1,158 @@
+	}
+
+	@Override
+	public PojoAutomaticIndexingTypeFilterContext exclude(Class<?> clazz) {
+		addIfNotPresentInOther(
+				contextProvider.indexedWithSuperTypesByExactClass().getOrFail( clazz ),
+				excludes,
+				includes
+		);
+		return this;
+	}
+
+	public HibernateOrmAutomaticIndexingTypeFilter createFilter() {
+		return createFilter( null );
+	}
+
+	public HibernateOrmAutomaticIndexingTypeFilter createFilter(PojoAutomaticIndexingTypeFilterHolder fallback) {
+		Set<PojoRawTypeIdentifier<?>> allIncludes = new HashSet<>();
+		Set<PojoRawTypeIdentifier<?>> allExcludes = new HashSet<>();
+		boolean allTypesProcessed = true;
+
+		for ( PojoTypeContext<?> typeContext : contextProvider.byEntityName().values() ) {
+			PojoRawTypeIdentifier<?> typedIdentifier = typeContext.typeIdentifier();
+
+			//.include(...) => included no matter what; subclasses included unless excluded explicitly in the same filter
+			//.exclude(...) => excluded no matter what; subclasses excluded unless included explicitly in the same filter
+			//.include(...) + .exclude(...) for the same entity in the same filter => exception
+			// neither .include(...) nor .exclude(...) => defer to the superclass inclusions/exclusions,
+			// then to the application filter, with the same behavior, and by default include
+			if ( excludes.contains( typedIdentifier ) ) {
+				allExcludes.add( typedIdentifier );
+			}
+			else if ( includes.contains( typedIdentifier ) ) {
+				allIncludes.add( typedIdentifier );
+			}
+			else {
+				Optional<Class<?>> closestInclude = findClosestClass( typedIdentifier.javaClass(), includes );
+				Optional<Class<?>> closestExclude = findClosestClass( typedIdentifier.javaClass(), excludes );
+
+				if ( closestInclude.isPresent() && closestExclude.isPresent() ) {
+					// if include is a subclass of exclude - then include is more specific, and we allow indexing:
+					if ( closestExclude.get().isAssignableFrom( closestInclude.get() ) ) {
+						allIncludes.add( typedIdentifier );
+					}
+					else {
+						allExcludes.add( typedIdentifier );
+					}
+				}
+				else if ( closestExclude.isPresent() ) {
+					allExcludes.add( typedIdentifier );
+				}
+				else if ( closestInclude.isPresent() ) {
+					allIncludes.add( typedIdentifier );
+				}
+				else {
+					// if we don't find either include or exclude then we will defer the decision to either app filter or to a default (include)
+					// but that will happen in the filter.
+					allTypesProcessed = false;
+				}
+			}
+		}
+
+		return HibernateOrmAutomaticIndexingTypeFilter.create(
+				fallback,
+				Collections.unmodifiableSet( allIncludes ),
+				Collections.unmodifiableSet( allExcludes ),
+				allTypesProcessed
+		);
+	}
+
+	private Optional<Class<?>> findClosestClass(Class<?> current, Set<PojoRawTypeIdentifier<?>> collection) {
+		Class<?> closest = Object.class;
+		for ( PojoRawTypeIdentifier<?> identifier : collection ) {
+			if ( identifier.javaClass().isAssignableFrom( current ) ) {
```

Comment:
You can't do that kind of things with named types (see `org.hibernate.search.integrationtest.mapper.orm.dynamicmap.DynamicMapBaseIT`).

Really, you need information that's only available on startup in order to handle that kind of things.

Please reconsider relying on a map `type => subtypes` built on startup like `org.hibernate.search.mapper.pojo.mapping.impl.PojoTypeManagerContainer#indexedBySuperType`, as I suggested some time ago :)

By the way, you might be able to implement this class in the `pojo-base` module, which would make all this available to the pojo-standalone module. I don't know if it's worth the trouble though?